clients/src/test/java/org/apache/kafka/common/config/provider/DirectoryConfigProviderTest.java

```diff
@@ -22,57 +22,67 @@
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.Locale;
+import java.util.Map;
 import java.util.ServiceLoader;
 import java.util.Set;
 import java.util.stream.StreamSupport;
 
 import static java.util.Arrays.asList;
+
+import static org.apache.kafka.common.config.provider.DirectoryConfigProvider.ALLOWED_PATHS_CONFIG;
 import static org.apache.kafka.test.TestUtils.toSet;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 public class DirectoryConfigProviderTest {
 
     private DirectoryConfigProvider provider;
+    @TempDir
     private File parent;
-    private File dir;
-    private File bar;
-    private File foo;
-    private File subdir;
-    private File subdirFile;
-    private File siblingDir;
-    private File siblingDirFile;
-    private File siblingFile;
-
-    private static File writeFile(File file) throws IOException {
-        Files.write(file.toPath(), file.getName().toUpperCase(Locale.ENGLISH).getBytes(StandardCharsets.UTF_8));
-        return file;
+    private String dir;
+    private final String bar = "bar";
+    private final String foo = "foo";
+    private String subdir;
+    private final String subdirFileName = "subdirFile";
+    private String siblingDir;
+    private final String siblingDirFileName = "siblingDirFile";
+    private final String siblingFileName = "siblingFile";
+
+    private static Path writeFile(Path path) throws IOException {
+        return Files.write(path, String.valueOf(path.getFileName()).toUpperCase(Locale.ENGLISH).getBytes(StandardCharsets.UTF_8));
     }
 
     @BeforeEach
     public void setup() throws IOException {
         provider = new DirectoryConfigProvider();
         provider.configure(Collections.emptyMap());
+
         parent = TestUtils.tempDirectory();
```

Comment:
This is unnecessary now with the annotation. `@TempDir` behaves like `@Mock`, in that the field is filled out by the test framework before the test execution starts.